### PR TITLE
Dbms session 2

### PR DESCRIPTION
**Pull request by meta-ishita-mathur**

### CODE REVIEW

File: DBMS-Session-2/ActionsOnTable.sql [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+LEFT JOIN Category P ON P.CategoryID = C.Parent_CategoryID 
+Order BY P.Category_Name, C.Category_Name;
+
+#Display Id, Title, Parent Category Title of all the leaf Categories 
+#(categories which are not parent of any other category)
+SELECT R.CategoryID, R.Category_Name, IFNULL(S.Category_Name, "Leaf Category") AS Leaf_Category
+FROM Category R
+LEFT JOIN Category S ON R.CategoryID = S.Parent_CategoryID
+WHERE R.CategoryID NOT IN
+(SELECT DISTINCT C.CategoryID FROM Category C INNER JOIN Category P ON C.CategoryID = P.Parent_CategoryID);
```
Comment (sandeep-sky) [inline DBMS-Session-2/ActionsOnTable.sql]: Please elaborate this query in meeting.

File: DBMS-Session-2/ActionsOnTable.sql [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+
+#Display Product Title, Price & Description which falls into particular category Title 
+#(i.e. “Mobile”)
+SELECT Product_Name, Product_Price, Discription FROM Product 
+WHERE ProductID IN (SELECT ProductID FROM Product_Category_Bridge WHERE CategoryID IN
+(SELECT CategoryID FROM Category WHERE Category_Name = "MOBILE"));
```
Comment (sandeep-sky) [inline DBMS-Session-2/ActionsOnTable.sql]: Why third level sub query? Please elaborate this query in meeting.

File: DBMS-Session-2/ActionsOnTable.sql [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+
+#Display the list of Products whose Quantity on hand (Inventory) is under 50.
+SELECT Product_Name FROM Product WHERE Product_Quantity < 5;
+
+#Increase the Inventory of all the products by 100
+SET SQL_SAFE_UPDATES = 0;
```
Comment (sandeep-sky) [inline DBMS-Session-2/ActionsOnTable.sql]: What is the use of `SQL_SAFE_UPDATES`?

File: DBMS-Session-2/ActionsOnTable.sql [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+UPDATE Product SET Product_Quantity = Product_Quantity + 100;
+SELECT ProductID, Product_Name, Product_Quantity FROM Product;
+
+#Assignment 3
+
+#Display Recent 50 Orders placed(Id, Order Date, Order Total)
+SELECT OrderID, Order_Date, Total_Price 
+FROM Order_Details
+ORDER BY Order_Date DESC LIMIT 50;
```
Comment (sandeep-sky) [inline DBMS-Session-2/ActionsOnTable.sql]: Need to discuss all the queries of `Assignment 3` in next meeting.

File: DBMS-Session-2/StoreFront.sql [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+    CategoryID int NOT NULL AUTO_INCREMENT,
+    Category_Name varchar(100) NOT NULL,
+    Parent_CategoryID int,
+    PRIMARY KEY(CategoryID)
+);
+    
+CREATE TABLE Product (
+    ProductID int NOT NULL AUTO_INCREMENT,
+    Product_Name varchar(100) NOT NULL,
+    Product_Quantity int,
```
[the remark this answers is not in the document]
Comment (meta-ishita-mathur) [reply DBMS-Session-2/StoreFront.sql]: DONE

File: DBMS-Session-2/ActionsOnTable.sql [hunk 6]
```diff
@@ -0,0 +1,106 @@
...
+
+#Assignment 2
+
+# Q.Display Id, Title, Category Title, Price of the products which are Active and
+#  recently added products should be at top.
+SELECT DISTINCT Product.ProductID, Product_Name, Category_Name, Product_Price, Product_Issue_Date
+FROM Product, Category, Product_Category_Bridge
+WHERE Product.ProductID = Product_Category_Bridge.ProductID
+AND Product_Category_Bridge.CategoryID = Category.CategoryID
+ORDER BY Product_Issue_Date DESC;
```
[the remark this answers is not in the document]
Comment (meta-ishita-mathur) [reply DBMS-Session-2/ActionsOnTable.sql]: done

File: DBMS-Session-2/ActionsOnTable.sql [hunk 7]
```diff
@@ -0,0 +1,106 @@
...
+
+#Display the list of products which don't have any images.
+SELECT ProductID, Product_Name, DIscription, Product_Price, Product_Quantity, Product_Issue_Date 
+FROM PRODUCT 
+WHERE Product.ProductID NOT IN (SELECT ProductID FROM Product_Images);
```
[the remark this answers is not in the document]
Comment (meta-ishita-mathur) [reply DBMS-Session-2/ActionsOnTable.sql]: DONE

File: DBMS-Session-2/ActionsOnTable.sql (hunk 1 above)
Comment (meta-ishita-mathur) [reply DBMS-Session-2/ActionsOnTable.sql]: DONE

File: DBMS-Session-2/ActionsOnTable.sql (hunk 2 above)
Comment (meta-ishita-mathur) [reply DBMS-Session-2/ActionsOnTable.sql]: Resolved it.

File: DBMS-Session-2/ActionsOnTable.sql (hunk 3 above)
Comment (meta-ishita-mathur) [reply DBMS-Session-2/ActionsOnTable.sql]: discussed in meeting

File: DBMS-Session-2/ActionsOnTable.sql (hunk 4 above)
Comment (meta-ishita-mathur) [reply DBMS-Session-2/ActionsOnTable.sql]: done

**Pull request closed without merging**